clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -280,7 +295,27 @@ void Sema::checkSYCLDeviceVarDecl(VarDecl *Var) {
   SourceRange Loc = Var->getLocation();
   llvm::DenseSet<QualType> Visited;
 
-  checkSYCLVarType(*this, Ty, Loc, Visited);
+  checkSYCLVarType(*this, Ty, Loc, Visited, false);
+}
+
+static void checkAccessorVarType(Sema &S, QualType Ty, SourceRange Loc) {
+  assert(Util::isSyclAccessorType(Ty) &&
+         "Should only be called on SYCL accessor types.");
+
+  llvm::DenseSet<QualType> Visited;
+  const RecordDecl *RecTy = Ty->getAsRecordDecl();
+  for (const auto &Field : RecTy->fields()) {
```

Comment:
Good catch. So in the case where there is a buffer of structs (that have a prohibited type member), this might not catch. I will check it.  If necessary, I should be able to move the accessor type check into the main type checking routine, since that one is recursive and specifically does enforce typing on struct members.

Reply:
I checked and yes it handles the case of a structure that contains prohibited types.

Reply:
test added